### PR TITLE
Feature/query update animals

### PR DESCRIPTION
**Pull request by eduardosancho**
# What was done in this PR by @eduardosancho 🎃 : 

- [x] Add the species column to the animals' table. 🐕 

- [x] Use transactions to assign a species to every animal. 👍 

- [x] Use transactions and SAVEPOINTs to safely delete items from __animals__. 🧷 

- [x] Use Aggregates and GROUP BY to extract filtered data from __animals__. ⚙️ 

dumbo

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,39 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT * FROM animals;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+SELECT * FROM animals;
+COMMIT;
+
+SELECT COUNT(name) FROM animals;
+SELECT COUNT(name) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
+SELECT MAX(weight_kg), MIN(weight_kg) FROM animals GROUP BY species;
```
Comment (andersonlebon) [inline queries.sql]: - [x] Nice job :clap: on getting the MIN and MAX weight however, you forgot to select the species column. Here is the deference :point_down: 

#### your  output commend 
image

#### the output selected with species
image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @eduardosancho 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## Required Changes ♻️

- [x] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (eduardosancho) [reply queries.sql]: Thank you for providing this clue, it was super useful!

File: data.sql [hunk 2]
```diff
@@ -4,3 +4,11 @@ INSERT INTO animals VALUES(1, 'Agumon', '2020-02-03', 0, true, 10.23);
 INSERT INTO animals VALUES(2, 'Gabumon', '2018-11-15', 2, true, 8);
 INSERT INTO animals VALUES(3, 'Pikachu', '2021-01-07', 1, false, 15.04);
 INSERT INTO animals VALUES(4, 'Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals VALUES(5, 'Charmander', '2020-02-08', 0, false, -11);
```
Comment (DammyShittu) [inline data.sql]: [OPTIONAL]
- I noticed that you're hardcoding values into the `id` column. What do you think about not having to bother about what number is next to be inserted into the `id` column?(imagine you had ten million rows) 😉
You can add the `GENERATED ALWAYS AS IDENTITY` constraint to the `id` column. This ensures that sequential integers that cannot be altered are generated each time a record is being created. I hope it makes sense. 😄 
You can go through this [article](https://www.sqltutorial.org/sql-identity/#:~:text=The%20GENERATED%20ALWAYS%20generates%20sequential,integers%20for%20the%20identity%20column.).

**Review by DammyShittu — APPROVED** [review]
Hi @eduardosancho, 👋 

# APPROVED 🟢 

Great implementation of all the requirements in this milestone and the change suggested by the previous reviewer. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

# OPTIONAL SUGGESTION

_Kindly check the comment below._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 2 above)
Comment (eduardosancho) [reply data.sql]: Enormous thanks for this tip.
I'm loving SQL 😄

**Pull request merged**